File: core/operator.ts
import Anthropic from '@anthropic-ai/sdk';

export interface OperatorConfig {
  anthropicApiKey: string;
  facilitatorUrl: string;
}

export class AtlasOperator {
  private client: Anthropic;
  private config: OperatorConfig;
  private conversationHistory: Array<{ role: string; content: string }> = [];

  constructor(config: OperatorConfig) {
    this.config = config;
    this.client = new Anthropic({ apiKey: config.anthropicApiKey });
  }

  async chat(options: { message: string; context?: Record<string, any> }): Promise<any> {
    const messages = [
      ...this.conversationHistory,
      { role: 'user', content: options.message },
    ];

    const systemPrompt = this.buildSystemPrompt(options.context || {});

    const response = await this.client.messages.create({
      model: 'claude-3-5-opus-20241022',
      max_tokens: 2048,
      system: systemPrompt,
      messages: messages,
    });

    const content = response.content[0].type === 'text' ? response.content[0].text : '';
    
    this.conversationHistory.push({ role: 'user', content: options.message });
    this.conversationHistory.push({ role: 'assistant', content });

    return { message: content };
  }

  async discoverServices(filters: Record<string, any> = {}): Promise<any[]> {
    const url = new URL(`${this.config.facilitatorUrl}/discovery/resources`);
    Object.entries(filters).forEach(([key, value]) => {
      url.searchParams.set(key, String(value));
    });

    const response = await fetch(url.toString());
    const data = await response.json();
    return data.resources || [];
  }

  private buildSystemPrompt(context: Record<string, any>): string {
    return `You are Atlas Operator, the AI control plane for Atlas402.
Network: ${context.network || 'base'}
Budget: ${context.budget || 'unlimited'}`;
  }
}

export default AtlasOperator;

File: core/operator.py
from typing import Optional, Dict, Any, List
from anthropic import Anthropic

class AtlasOperator:
    def __init__(
        self,
        anthropic_api_key: str,
        facilitator_url: str
    ):
        self.client = Anthropic(api_key=anthropic_api_key)
        self.facilitator_url = facilitator_url
        self.conversation_history: List[Dict[str, str]] = []
        
    async def chat(
        self,
        message: str,
        context: Optional[Dict[str, Any]] = None
    ) -> Dict[str, str]:
        messages = [
            {'role': msg['role'], 'content': msg['content']}
            for msg in self.conversation_history
        ]
        messages.append({'role': 'user', 'content': message})
        
        system_prompt = self._build_system_prompt(context or {})
        
        response = self.client.messages.create(
            model='claude-3-5-opus-20241022',
            max_tokens=2048,
            system=system_prompt,
            messages=messages
        )
        
        content = response.content[0].text
        
        self.conversation_history.append({'role': 'user', 'content': message})
        self.conversation_history.append({'role': 'assistant', 'content': content})
        
        return {'message': content}
    
    async def discover_services(self, filters: Dict[str, Any] = None) -> List[Dict[str, Any]]:
        import aiohttp
        url = f"{self.facilitator_url}/discovery/resources"
        async with aiohttp.ClientSession() as session:
            async with session.get(url, params=filters or {}) as response:
                data = await response.json()
                return data.get('resources', [])
    
    def _build_system_prompt(self, context: Dict[str, Any]) -> str:
        return f"""You are Atlas Operator, the AI control plane for Atlas402.
Network: {context.get('network', 'base')}
Budget: {context.get('budget', 'unlimited')}"""




